handler/handler.go

```diff
@@ -297,71 +292,63 @@ func (h *InstanceComplete) GetS3ContentLength(ctx context.Context, instanceID st
 // UpdateInstance updates the instance downlad CSV link using dataset API PUT /instances/{id} endpoint
 // if the instance is published, then the s3Url will be set as public link and the instance state will be set to published
 // otherwise, a private url will be generated and the state will not be changed
-func (h *InstanceComplete) UpdateInstance(ctx context.Context, instanceID string, size int, isPublished bool, s3Url string) error {
+func (h *InstanceComplete) UpdateInstance(ctx context.Context, e *event.ExportStart, size int, isPublished bool, s3Url string) error {
+	csvDownload := dataset.Download{
+		Size: fmt.Sprintf("%d", size),
+		URL: fmt.Sprintf("%s/downloads/datasets/%s/editions/%s/versions/%s.csv",
+			h.cfg.DownloadServiceURL,
+			e.DatasetID,
+			e.Edition,
+			e.Version,
+		),
+	}
+
 	if isPublished {
-		update := dataset.UpdateInstance{
-			Downloads: dataset.DownloadList{
-				CSV: &dataset.Download{
-					URL:    s3Url,                   // Public URL, as returned by the S3 uploader
-					Public: s3Url,                   // Public URL, as returned by the S3 uploader
-					Size:   fmt.Sprintf("%d", size), // size of the file in number of bytes
-				},
-			},
-		}
-		if _, err := h.datasets.PutInstance(ctx, "", h.cfg.ServiceAuthToken, "", instanceID, update, headers.IfMatchAnyETag); err != nil {
-			return fmt.Errorf("error during put instance: %w", err)
-		}
-		return nil
+		csvDownload.Public = s3Url
+	} else {
+		csvDownload.Private = s3Url
 	}
 
-	update := dataset.UpdateInstance{
-		Downloads: dataset.DownloadList{
-			CSV: &dataset.Download{
-				Size: fmt.Sprintf("%d", size),                                  // size of the file in number of bytes
-				URL:  generatePrivateURL(h.cfg.DownloadServiceURL, instanceID), // Private downloadService URL
-			},
+	log.Info(ctx, "updating dataset api with download link", log.Data{
+		"is_published": isPublished,
+		"csv_download": csvDownload,
+	})
+
+	versionUpdate := dataset.Version{
+		Downloads: map[string]dataset.Download{
+			"CSV": csvDownload,
 		},
 	}
-	if _, err := h.datasets.PutInstance(ctx, "", h.cfg.ServiceAuthToken, "", instanceID, update, headers.IfMatchAnyETag); err != nil {
-		return fmt.Errorf("error during put instance: %w", err)
+
+	err := h.datasets.PutVersion(
+		ctx, "", h.cfg.ServiceAuthToken, "", e.DatasetID, e.Edition, e.Version, versionUpdate)
```

Comment:
Bracket should close on following line (or have arguments stacked vertically). Could also use `if err := ` initialisation